### DISCUSSION TITLE
K8 Readiness Liveness

### DISCUSSION THREAD

**Discussion by jmarcos-cano** [BretFisher/ama#166] [Q&A]
Hello Bret, while adding a [readiness & liveness](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) probes to our K8 deployments should we even consider:

- code endpoints (/liveness & /readiness) in our app to respond to this K8 specific scenario?
- if so, what is the recommendations for each of those?
- or a ` /health => response(200)` is enough? like [this](https://github.com/BretFisher/node-docker-good-defaults/blob/master/index.js#L53)

thanks in advance

**Comment by BretFisher** [comment]
It's always the response code that you want to go for, and if you're advanced enough to properly respond to liveness and readiness, knowing the differences, and know you'll use Kubernetes, then sure build those into your apps.  Docker Swarm currently only supports liveness.  Making the URL's "kube friendly" certainly helps make it clear which to use for what :)
